google-cloud-spanner/lib/google/cloud/spanner/database/job.rb

```diff
@@ -122,6 +122,32 @@ def error?
           end
 
           ##
+          # The status code and message if the operation associated with this
+          # job produced an error, returned as an error object.
+          #
+          # @return [Google::Cloud::Error, nil] An error object containing the
+          #   status code and message, or `nil` if no error occurred.
+          #
+          # @example
+          #   require "google/cloud/spanner"
+          #
+          #   spanner = Google::Cloud::Spanner.new
+          #
+          #   job = spanner.create_database "my-instance",
+          #                                 "my-new-database"
+          #
+          #   job.error? # true
+          #
+          #   error = job.error
+          #
+          def error
+            return nil unless error?
+            fail GRPC::BadStatus.new @grpc.error.code, @grpc.error.message
```

Comment:
While I'm not in favor of leaking the low-level API, I am beginning to think a status object is more appropriate than returning an un-raised Error, which just seems a little off-pattern.